source/blog/2015-06-12-ember-1-13-0-released.md

```diff
@@ -0,0 +1,578 @@
+granular observation was computationally expensive to initialize and maintain
+for a large set of values.
+
+Glimmer replaces this approach with a value-diffing strategy.
+During render, each occurrence of dynamic content (for example a component,
+and each helper, or a data binding) generates a value representing its
+render state. Upon rerender, this tree of values is walked. Clean nodes are
+presumed to already be correctly rendered, dirtied nodes have their value
+recomputed and children values (which may or may not have dirtied) walked.
+
+The result is a stunning improvement in many rerender cases. This animated
+gif shows rerender performance of Ember 1.12 on the left, and Ember 1.13
+on the right:
+
+![](ANIMATED GIF)
+
+Thanks to [@wycats](https://twitter.com/wycats) and [@tomdale](https://twitter.com/tomdale)
+for their continued focus on improving
+Glimmer's performance, and to [LinkedIn](http://www.linkedin.com) and
+[Bustle](http://www.bustle.com/) for their
+generous sponsorship of this work.
+
+#### attrs, mut and auto-mut
+
+In Ember 1.13, components will now have an `attrs` property set with the
+passed values. For example:
+
+```hbs
+{{my-component name="Susan"}}
+```
+
+```js
+// app/components/my-component.js
+import Ember from "ember";
+
+export default Ember.Component.extend({
+  click() {
+    console.log(this.getAttr('name')); // -> logs "Susan"
+    console.log(this.attrs.name); // -> logs "Susan"
+    // The old behavior still works:
+    console.log(this.get('name')); // -> logs "Susan"
+  }
+});
+```
+
+The Glimmer rendering engine uses one-way binding at its core. In order to
+implement the current two-way binding behavior of components, Glimmer requires
+the addition of mutable values. We call these `mut` objects, since the `mut`
+helper will be used to create them once angle components land.
+
+`mut` objects represent a mutable value. They have a current value, and a
+function that can update that value.
+
+* `someMut.value` is the current value of the mutable value
+* `someMut.update(newValue)` updates the value of the mutable value
+
+Conceptually, a `mut` simply sets the new value to whatever it is wrapping
+then triggers a rerender at whatever template
+created the `mut`.
+
+Curly components use "auto-mut" to emulate two-way binding. Any bound attribute
+is wrapped as a `mut` object before being set on `attrs`. For example, making
+the `name` attribute bound results in a `mut` object being set to `attrs.name`:
+
+```hbs
+{{my-component name=model.name}}
+```
+
+```js
+// app/components/my-component.js
+import Ember from "ember";
+
+export default Ember.Component.extend({
+  click() {
+    // getAttr always returns a value, regardless of if an attr is a mut
+    console.log(this.getAttr('name')); // -> logs model.name's value
+    // Note that `name` is a mut object:
+    console.log(this.attrs.name.value); // -> logs model.name's value
+    // The old behavior still works:
```

Comment:
Not deprecated. With the intent that most components will migrate to angles anyway, and making people update the curly components just to upgrade to angles later on is painful. I think that is the justification, should be discussed with others though.